linode/manager
# test: Improve unit test stability when running locally

## Description 📝
When running tests locally on `develop`, I'm getting anywhere between 3-8 failures each run. We additionally had a flaky unit test in CI during our last release which required a few re-runs to unblock the fire chief. This PR aims to address these failures and clean up a few warning messages.

Many of the failures are mitigated by increasing test timeouts -- I investigated the failures and the corresponding components and found that the tests and components were behaving exactly as expected, but when running alongside the rest of the test suite, the tests' performance drops dramatically (e.g. from 1-2s to 7-8s, or more). Going to post some more details as comments to this PR.

## Changes  🔄
- Upgrade Vitest from 1.2.x to 1.3.x.
- Increase test timeouts for flaky tests in `CreateCertificateDrawer.test.tsx`, `CreateAPITokenDrawer.test.tsx`, and `ImageAndPassword.test.tsx`
- Remove calls to `act()` tests to resolve some failures and warnings
  - In most cases, we were using React Testing Library functions that call `act()` inside of our own `act()` statements which triggered the issues (`userEvent.*`, `findBy*`, etc.)

## Target release date 🗓️
It'd be great to get this in by the next release just to reduce any risk of a unit test failure hindering the release process, but aside from that there's really no urgency here.

## How to test 🧪
This is tricky because most of these failures 1) only occur when running locally, 2) only occur when running the entire test suite (or some substantial portion of the test suite), and 3) seem to be performance related and therefore may vary from machine-to-machine.

The best way to confirm whether things are stable is to run the test suite locally:

```bash
yarn && yarn test
```

## As an Author I have considered 🤔

*Check all that apply*
